source_gen/lib/src/constants/reviver.dart

```diff
@@ -0,0 +1,194 @@
+      return reader.literalValue;
+    } else if (reader.isType) {
+      return reader.typeValue;
+    } else if (reader.isSymbol) {
+      return reader.symbolValue;
+    } else {
+      final decl = classMirror;
+      if (decl.isEnum) {
+        final values = decl.getField(const Symbol('values')).reflectee as List;
+        return values[reader.objectValue.getField('index')!.toIntValue()!];
+      }
+
+      final pv = positionalValues;
+      final nv = namedValues;
+
+      return decl
+          .newInstance(Symbol(reader.revive().accessor), pv, nv)
+          .reflectee;
+    }
+  }
+
+  dynamic get primativeValue {
+    if (reader.isNull) {
+      return null;
+    } else if (reader.isBool) {
+      return reader.boolValue;
+    } else if (reader.isDouble) {
+      return reader.doubleValue;
+    } else if (reader.isInt) {
+      return reader.intValue;
+    } else if (reader.isString) {
+      return reader.stringValue;
+    }
+  }
+
+  List<dynamic> get positionalValues => reader
+      .revive()
+      .positionalArguments
+      .map(
+        (value) => Reviver.fromDartObject(value).toInstance(),
+      )
+      .toList();
+
+  Map<Symbol, dynamic> get namedValues => reader.revive().namedArguments.map(
+        (key, value) {
+          final k = Symbol(key);
+          final v = Reviver.fromDartObject(value).toInstance();
+          return MapEntry(k, v);
+        },
+      );
+
+  ClassMirror get classMirror {
+    final revivable = reader.revive();
+
+    // Flatten the list of libraries
+    final entries = Map.fromEntries(currentMirrorSystem().libraries.entries)
+        .map((key, value) => MapEntry(key.pathSegments.first, value));
+
+    // Grab the library from the system
+    final libraryMirror = entries[revivable.source.pathSegments.first];
+    if (libraryMirror == null || libraryMirror.simpleName == Symbol.empty) {
+      throw Exception('Library missing');
+    }
+
+    // Determine the declaration being requested. Split on . when an enum is passed in.
+    var declKey = Symbol(revivable.source.fragment);
+    if (reader.isEnum) {
+      // The accessor when the entry is an enum is the ClassName.value
+      declKey = Symbol(revivable.accessor.split('.')[0]);
+    }
+
+    final decl = libraryMirror.declarations[declKey] as ClassMirror?;
+    if (decl == null) {
+      throw Exception('Declaration missing');
+    }
+    return decl;
+  }
+
+  List<T>? toTypedList<T>(T t) => reader.listValue
+      .map((e) => Reviver.fromDartObject(e).toInstance() as T)
+      .toList() as List<T>?;
+
+  Map<KT, VT>? toTypedMap<KT, VT>(KT kt, VT vt) => reader.mapValue.map(
+        (key, value) => MapEntry(
+          Reviver.fromDartObject(key).toInstance() as KT,
+          Reviver.fromDartObject(value).toInstance() as VT,
+        ),
+      ) as Map<KT, VT>?;
+
+  Set<T>? toTypedSet<T>(T t) => reader.setValue
+      .map((e) => Reviver.fromDartObject(e).toInstance() as T)
+      .toSet() as Set<T>?;
+}
```

Comment:
Trying to leverage these results in the following stack trace:
```
dart:collection                                       MapBase.map
package:source_gen/src/constants/reviver.dart 178:68  Reviver.toTypedMap
package:source_gen/src/constants/reviver.dart 93:16   Reviver.toInstance
package:source_gen/src/constants/reviver.dart 142:51  Reviver.namedValues.<fn>
dart:collection                                       MapBase.map
package:source_gen/src/constants/reviver.dart 139:74  Reviver.namedValues
package:source_gen/src/constants/reviver.dart 109:18  Reviver.toInstance
package:source_gen/src/constants/reviver.dart 135:50  Reviver.positionalValues.<fn>
dart:_internal                                        ListIterable.toList
package:source_gen/src/constants/reviver.dart 137:8   Reviver.positionalValues
package:source_gen/src/constants/reviver.dart 108:18  Reviver.toInstance
test/constants/reviver_test.dart 56:38                main.<fn>.<fn>.<fn>.<fn>
```

This feels wrong. The reason I'm doing this is because there is a limitation of not being able to do:
```dart
final t = Reviver.fromDartObject(reader.listValue.first)
              .toInstance()
              .runtimeType;

return reder.listValue.map((e) => Reviver.fromDartObject(e).toInstance() as t).toList();
```